packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/CsvAsset.java

```diff
@@ -0,0 +1,33 @@
+package software.uncharted.terarium.hmiserver.models.dataservice;
+
+import lombok.Data;
+import lombok.experimental.Accessors;
+import lombok.extern.slf4j.Slf4j;
+import software.uncharted.terarium.hmiserver.annotations.TSModel;
+import software.uncharted.terarium.hmiserver.annotations.TSOptional;
+
+import java.io.Serializable;
+import java.util.*;
+
+@Data
+@Accessors(chain = true)
+@Slf4j
+@TSModel
+public class CsvAsset implements Serializable {
+	List<List<String>> csv;
+	@TSOptional
+	List<CsvColumnStats> stats;
+	List<String> headers;
+
+	public CsvAsset(List<List<String>> csv, List<CsvColumnStats> stats, List<String> headers){
+		this.csv = csv;
+		this.stats = stats;
+		this.headers = headers;
+	}
+
+	public CsvAsset(List<List<String>> csv, List<String> headers){
```

Comment:
Similarly here...
If `csv` and `headers` are always required to be present on this object, you can annotate those specific fields with `@NonNull`, and then add the annotation `@RequiredArgsConstructor` to the class itself. This will....

1) Add null checking to the all-args constructor
2) Create a second constructor for just `csv` and `headers` 

Again, both of these suggestions are optional - what you have here works fine - but Lombok can save you some typing :)

Reply:
Annotation magic scares cavemen like me